hudi-utilities/src/main/java/org/apache/hudi/utilities/UtilHelpers.java

```diff
@@ -122,6 +122,9 @@ public static Source createSource(String sourceClass, TypedProperties cfg, JavaS
                 HoodieDeltaStreamerMetrics.class},
             cfg, jssc, sparkSession, schemaProvider, metrics);
       } catch (HoodieException e) {
+        if (e.getCause() != null && !(e.getCause() instanceof NoSuchMethodException)) {
+          LOG.error("Could not load source class " + sourceClass, e);
```

Comment:
I would suggest we print this error message directly. No need to add this condition `e.getCause() != null && !(e.getCause() instanceof NoSuchMethodException)`. WDYT?

You just want to log more information about this exception, right? Just log it.

Reply:
Thanks for your review! I think we shouldn't print error msg of NoSuchMethodException, because it is noisy and user doesn't need to know that.

Reply:
Maybe some Source class just implement constructor of `(TypedProperties properties, JavaSparkContext sparkContext, SparkSession sparkSession, SchemaProvider schemaProvider)`, and if we don't judge, there are a lot of error messages.

Reply:
@Clcanny this sort of logic is very unconventional. when there is an error, we log it. that's simple and straightforward. if you find a lot of unwanted logs, then put a fix on the root cause, instead of putting a band-aid fix like this.